### ISSUE TITLE
Generate classes from GraphQL definitions

### ISSUE THREAD

**Issue by codepunkt** [graphql-dotnet/graphql-dotnet#317]
I wonder if it is possible to automatically generate `graphql-dotnet` classes (either default or [conventions](https://github.com/graphql-dotnet/conventions) style) based on schema and type definitions given in graphql language (`.gql` or `.graphql` files).

e.g. transform *this*
```graphql
type Query {
  # Fetch a list of users
  users: [User]!
}

type Mutation {
  # Sign up
  signUp(email: String!, password: String!): Auth

  # Sign in
  signIn(email: String!, password: String!): Auth
}

schema {
  query: Query
  mutation: Mutation
}
```

into `graphql-dotnet` usable classes.

Have you thought about that yet or do any other projects allow this?

Is there any other way to directly (re)use `.gql` or `.graphql` files? This is rather easy with [Apollo](http://dev.apollodata.com/) in the NodeJS ecosystem and seems like a large benefit to me. Looking at large APIs i'd like to build with `graphql-dotnet`, i'm not too pleased by having to write all that code by hand.

**Comment by tlil** [comment]
> Looking at large APIs i'd like to build with graphql-dotnet, i'm not too pleased by having to write all that code by hand.

That's been the motivation behind [graphql-dotnet/conventions](https://github.com/graphql-dotnet/conventions). Although it doesn't allow you to map a plain text schema file in like you describe, it allows you to express your schema more succinctly.

Using the GraphQL schema files like you describe is nice too though :) Dunno if @joemcbride has any plans in mind re that, but I have definitely been thinking about it. Not started anything yet though.

**Comment by joemcbride** [comment]
> Have you thought about that yet or do any other projects allow this?

I have thought about this.  Enabling this was one of my primary motivations for the "instance-based" api.  I have a spike of this somewhere.  What I'm not sure about is what the API should look like in the .NET world for the field resolvers.  Have ideas?

**Comment by codepunkt** [comment]
Actually, i can't help much with that (yet). I've just been doing .NET for the better part of 2 weeks so i'm actually glad i finally go a basic example running using my Unity DI setup.

Great to hear you've been thinking about this, though! Will keep an eye on this and might be able to comment or chip in with ideas once i'm more settled in the language and ecosystem!

**Comment by joemcbride** [comment]
Here is a start for this: https://github.com/graphql-dotnet/graphql-dotnet/pull/362

**Comment by joemcbride** [comment]
#362 is merged and there is an alpha of 2.0 available.  Going to close this as complete.
